infernyx/rules.py

```diff
@@ -359,11 +445,33 @@ def tag_results(suffix, job):
                 parts_preprocess=[parse_distinct])
         },
 
-        # note that this rule_cleanup will be obsolete after the PR https://github.com/chango/inferno/pull/23
-        # is merged and released, then only the 'result_tag' below will be required
-        # result_tag='incoming:site_tuples',
-        rule_cleanup=partial(tag_results, 'incoming:site_tuples:'),
-        save=True,
-        no_purge=True,
+        result_tag='incoming:site_tuples',
+    ),
+    InfernoRule(
+        name='ip_click_counter',
+        source_tags=['incoming:impression'],
+
+        # run this job every day at 2am on yesterday's data
+        day_range=1,
+        day_offset=1,
+        time_delta={'oclock': 2},
+
+        map_input_stream=chunk_json_stream,
+        parts_preprocess=[clean_data, parse_date, parse_ip_clicks],
+        parts_postprocess=[partial(filter_clicks, click_threshold=50, impression_threshold=750)],
+        result_processor=partial(report_suspicious_ips,
+                                 host=RS_HOST,
+                                 port=RS_PORT,
+                                 database=RS_DB,
+                                 user=RS_USER,
+                                 password=RS_PASSWORD,
+                                 bucket_name=RS_BUCKET),
+        partitions=32,
+        sort_buffer_size='25%',
+        combiner_function=combiner,
+        key_parts=['date', 'ip'],
+        value_parts=['impressions', 'clicks'],
+        column_mappings={'impressions': None, 'clicks': None},
```

Comment:
Inferno used to ignore columns when their column mappings were set to
None.  That isnt happening anymore!

I'll submit a fix.

On Thu, May 28, 2015 at 2:11 PM, Nan Jiang <notifications@github.com> wrote:

> In infernyx/rules.py
> <https://github.com/tspurway/infernyx/pull/20#discussion_r31261420>:
>
> > +        map_input_stream=chunk_json_stream,
> > +        parts_preprocess=[clean_data, parse_date, parse_ip_clicks],
> > +        parts_postprocess=[partial(filter_clicks, click_threshold=50, impression_threshold=750)],
> > +        result_processor=partial(report_suspicious_ips,
> > +                                 host=RS_HOST,
> > +                                 port=RS_PORT,
> > +                                 database=RS_DB,
> > +                                 user=RS_USER,
> > +                                 password=RS_PASSWORD,
> > +                                 bucket_name=RS_BUCKET),
> > +        partitions=32,
> > +        sort_buffer_size='25%',
> > +        combiner_function=combiner,
> > +        key_parts=['date', 'ip'],
> > +        value_parts=['impressions', 'clicks'],
> > +        column_mappings={'impressions': None, 'clicks': None},
>
> This line appears to be redundant, inferno will ignore the column mapping
> if its value is None
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tspurway/infernyx/pull/20/files#r31261420>.
>